### PR TITLE
proxy: add response API tests in proxyunits.t

### PR DESCRIPTION
**Pull request by feihu-stripe**

### CODE REVIEW

File: vendor/mcmc/mcmc.c [hunk 1]
```diff
@@ -251,6 +252,8 @@ static int _mcmc_parse_response(mcmc_ctx_t *ctx, mcmc_resp_t *r) {
                             } else {
                                 r->vlen_read = buffer_remain;
                             }
+                            // remove vsize and any leading space from response length.
+                            l -= (n - cur) / sizeof(char);
```
Comment (dormando) [inline vendor/mcmc/mcmc.c:256]: sizeof(char) is defined as 1 in the C spec. can skip the division.

**Comment by dormando** [comment]
So, mcmc lives here: https://github.com/dormando/mcmc - with its own unit tests. I can port this over and add the tests but if you want to try go ahead.

when I make changes to mcmc I do them in order:
- change upstream
- individual commit updating vendor/mcmc
- individual commit utilizing the updated mcmc

it lets me see historically when vendor stuff changes specifically.

**Comment by feihu-stripe** [comment]
Sure I can give it a try.

**Comment by dormando** [comment]
might be nice in the meantime if you separate out the working tests so I can upstream those. you can leave the broken test in with a skip marker and repair it afterward.

**Comment by feihu-stripe** [comment]
will do

**Comment by feihu-stripe** [comment]
mcmc fix moved to https://github.com/dormando/mcmc/pull/3

PR should now be mergable.

File: t/proxyunits.t [hunk 2]
```diff
@@ -680,15 +680,164 @@ check_sanity($ps);
...
+            be_send => {0 => ["HD\r\n"]},
+            ps_recv => ["HD\r\n"],
+        );
+    };
+
+
+    subtest 'response:hit()' => sub {
+        # ps_recv must not receive an error
+        proxy_test(
+            ps_send => "mg /response/ok\r\n",
```
Comment (dormando) [inline t/proxyunits.t:712]: double testing ok() instead of hit here.

should also test cases where ok/hit are false work properly.

**Comment by dormando** [comment]
I think the rest looks good but kinda groggy and need to think more.

for the `hit()` and `ok()` tests there are a few backend conditions which can result in `ok()`. ideally we test more than one of them.

**Comment by feihu-stripe** [comment]
Added more cases for ok() and hit()

**Comment by dormando** [comment]
looks okay I think. I'm going to need to come back after a break for a final review though; if you think you're done feel free to squash it.

**Comment by feihu-stripe** [comment]
> looks okay I think. I'm going to need to come back after a break for a final review though; if you think you're done feel free to squash it.

Did what I can think of, but I guess it is never 100%. 

Squahsed.

**Comment by dormando** [comment]
Thanks. I think it's more important to round this out by ensuring we have `mcp.pool` two-arg tests now. We should be periodically revisiting the tests and adding layers to it instead of trying to waterfall full scenarios anyway.

**Comment by feihu-stripe** [comment]
>  we have mcp.pool two-arg tests now

sure will take a look

**Comment by dormando** [comment]
there're a bunch of examples commented in t/startfile.lua fwiw.

**Comment by dormando** [comment]
merged it. didn't feel like the comment was a blocker.

**Comment by dormando** [comment]
Released in 1.6.21

**Pull request closed without merging**